src/compiler/checker.ts

```diff
@@ -18219,6 +18219,22 @@ namespace ts {
                     return result;
                 }
 
+                if (isNonAugmentingArraySubtype(source) && isArrayType(target)) {

```

Comment:
What if they're both non-augmenting array subtypes?

Reply:
We could relate them; it's just not as common in practice. This comes back to my "we could generalize this more" remark in my prior comment - this doesn't really need to be limited to just `Array`s either.